client/flutter/lib/module_home/ui/state/home_loaded_state.dart

```diff
@@ -81,22 +83,41 @@ class HomeLoadedState extends HomeState {
         ListTile(leading: Icon(Icons.sort,color: Theme.of(context).primaryColor,),
           title: Text(S.of(context).categories,style:StyleText.categoryStyle,),
         ),
-        GridView.builder(
+        GridView(
           physics: ScrollPhysics(),
           shrinkWrap: true,
-          itemCount: 8,
           gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
               crossAxisCount: 2,
               mainAxisSpacing: 8.0,
               childAspectRatio: (MediaQuery.of(context).size.width / 2) / 135),
-          itemBuilder: (_, index) {
-            return ProductCard(title:'متجر ${index+1}', image: 'https://mk0kaleelao979sb1ktf.kinstacdn.com/wp-content/uploads/2020/03/Supermarket-conversation-in-Arabic.jpg');
-          },
+          children: getCategories(categories),
         ),
         SizedBox(
           height: 75,
         ),
       ],
     );
   }
-}
+
+  List<Widget> _getTopProducts(List<ProductModel> topProducts) {
+    if (topProducts.isEmpty) return [];
+    List<ProductCard> top = [];
+    topProducts.forEach((element) {
+      top.add(ProductCard(title:element.title,image: element.image,));
+    });
+    return top;
+  }
+
+  List<Widget> getCategories(List<StoreCategoryModel> categories) {
+    if (categories.isEmpty) return [];
+    List<ProductCard> cats = [];
```

Comment:
cats? mean like Meaw?

Reply:
we have to much categories so I thought if we have a nick name to it will be great

Reply:
we have to much categories so I thought if we have a nick name to it will be great